### PR TITLE
build: bump spark version to 3.4.3

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

**Review by viirya — COMMENTED** [review]
We only have diff for Spark 3.4.2. To build and test with Spark 3.4.3, you need to prepare diff for Spark 3.4.3.

**Review by parthchandra — APPROVED** [review]
LGTM  (pending ci)

**Review by kazuyukitanimura — APPROVED** [review]
LGTM

File: dev/diffs/3.4.3.diff [hunk 1]
```diff
@@ -0,0 +1,2564 @@
...
+diff --git a/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala b/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala
+index c595b50950b..6b60213e775 100644
+--- a/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala
++++ b/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala
+@@ -102,7 +102,7 @@ class SparkSession private(
+       sc: SparkContext,
+       initialSessionOptions: java.util.HashMap[String, String]) = {
+     this(sc, None, None,
+-      SparkSession.applyExtensions(
++      SparkSession.applyExtensions(sc,
```
Comment (viirya) [inline dev/diffs/3.4.3.diff:64]: I think the most significant change in this PR is this Spark code diff. Do you notice any obvious change to 3.4.2 diffs?

**Review by viirya — COMMENTED** [review]
Looks good to me as I quickly look through the 3.4.3 diffs. But wondering if @huaxingao finds any changes between 3.4.2 and 3.4.3 diffs we should give notice.

**Comment by huaxingao** [comment]
@viirya 
I didn't see any changes between 3.4.2.diff and 3.4.3.diff except some index changes, line number changes and the changes I added 
```
@@ -1128,7 +1130,7 @@ abstract class ParquetQuerySuite extends QueryTest with ParquetTest with SharedS
             .where(s"a < ${Long.MaxValue}")
             .collect()
         }
-        assert(exception.getCause.getCause.isInstanceOf[SchemaColumnConvertNotSupportedException])
+        assert(exception.getMessage.contains("Column: [a], Expected: bigint, Found: INT32"))
       }
     }
   }
```

**Comment by viirya** [comment]
> @viirya I didn't see any changes between 3.4.2.diff and 3.4.3.diff except some index changes, line number changes and the changes I added

Okay, that's great. I thought that there might be some conflicts between 3.4.2 and 3.4.3 tests, but seems we don't have much changes between the two patch versions.

**Comment by huaxingao** [comment]
Thanks @kazuyukitanimura @parthchandra @viirya

**Pull request merged**